Modified TextField docs - Replaced 'labelText' to 'hintText' in code snippet

vpaladino778 opened the pull request:
Below is a screenshot of the current state of the [TextField ](https://api.flutter.dev/flutter/material/TextField-class.html) docs. As you can see, the code snippet shows `labelText`, but the image that is attached shows `hintText`. This PR corrects that change so that the code snippet matches the image.


![image](https://user-images.githubusercontent.com/19626901/143135989-8105a9b0-9ffe-4cde-964b-812b2a1f4e23.png)

vpaladino778 committed:
Modified labelText to hintText

This modification now matches the image in the docs

Modified indexing of text_field examples be in order

Changed 'obscure' to 'obscured'

Fixed the indexing of example files based on conventions

Replaced snippet with dartpad example

Hixie commented:
test-exempt: only affects comments

That said, this is actually testable; if the sample snippet is turned into an actually piece of sample code (like most other samples) then it'd be run and analyzed and we would catch this kind of problem.

gspencergoog reviewed (approved):
![32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d](https://user-images.githubusercontent.com/8867023/37737237-66e9c9dc-2d10-11e8-8797-13a405eace49.png)

I'm happy to accept this update, but as @Hixie pointed out, it would be great to convert this to a "dartpad" sample.  That is a bit more work to do, though, so I understand if you're not interested.

vpaladino778 commented:
@gspencergoog  Hey Greg, I'd be happy to give it a go. I'll rerequest a review from you once I got it figured out.

gspencergoog commented:
OK, great!

Take a look at https://github.com/flutter/flutter/blob/master/examples/api/README.md for info on how to do it.

vpaladino778 commented:
I fixed the naming convention for the sample apps. Renamed `text_field.1.dart` to `text_field.0.dart`, and named the new sample app `text_field.1.dart`.

gspencergoog commented on packages/flutter/lib/src/material/text_field.dart:
````diff
@@ -172,22 +172,14 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
-///   ),
-/// )
-/// ```
+/// ** See code in examples/api/lib/material/text_field/text_field.1.dart **
````
Why isn't this one `text_field.0.dart`? It comes first in the comments, so it should have the lower index.

vpaladino778 replied:
Good point, I agree. I'll make the switch

gspencergoog commented on packages/flutter/lib/src/material/text_field.dart:
```diff
@@ -172,22 +172,14 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
 /// by the object.
 ///
-/// {@tool snippet}
+/// ## Obscure Input
```
```suggestion
/// ## Obscured Input
```

gspencergoog commented on packages/flutter/lib/src/material/text_field.dart:
```diff
@@ -172,22 +172,14 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
+/// ## Obscure Input
+///
+/// {@tool dartpad}
 /// This example shows how to create a [TextField] that will obscure input. The
```
```suggestion
/// This example shows how to create a [TextField] that will obscured input. The
```

gspencergoog commented on packages/flutter/lib/src/material/text_field.dart:
```diff
@@ -172,22 +172,14 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
+/// ## Obscured Input
+///
+/// {@tool dartpad}
+/// This example shows how to create a [TextField] that will obscured input. The
```
```suggestion
/// This example shows how to create a [TextField] that will obscure input. The
```

Sorry, I gave you a bad suggestion before.

gspencergoog reviewed (changes requested):
OK, this is great, now the only thing missing is a test for the new example.

You can see an example of what a test might look like here:
https://github.com/flutter/flutter/blob/master/examples/api/test/widgets/layout_builder/layout_builder.0_test.dart

Hixie commented:
@vpaladino778 Sorry to make you jump through all these hoops for a typo fix... is there any chance you are able to add a test like @gspencergoog suggested? No worries if not, we can take it from here if necessary.

vpaladino778 commented:
> @vpaladino778 Sorry to make you jump through all these hoops for a typo fix... is there any chance you are able to add a test like @gspencergoog suggested? No worries if not, we can take it from here if necessary.

Yeah i'm up for it. I had gotten started a while back but mostly forgot to finish it up. I can take a look this weekend and hopefully have the test working.

Hixie commented:
@vpaladino778 Just checking in, anything we can do to help?

Hixie commented:
cc @goderbauer

vpaladino778 commented:
> @vpaladino778 Just checking in, anything we can do to help?

Hey there, I can take a look at this tonight. Sorry for the delay, just started a new job and have been busy with that.  

Last time i tried getting the test working I had some issues with my local development environment. My Android studio only shows the 'java' folder and no files. And the dart file i have open has 'Target URI not found' errors all over it, also getting 'Pub get has not been run'. So i think something wacky is going on with my IDE. I'm going to try and update android studio and reimport the project. Let me know if you have any suggestions

Hixie commented:
@vpaladino778 congratulations on the new job! If you do have time to work on this please feel free to reach out on our [Discord](https://github.com/flutter/flutter/wiki/Chat) to get help with writing the test, the #hackers-new channel is the best place to reach out. Thanks!

Hixie commented:
@vpaladino778 I feel bad that we made you jump through all those hoops and as a result the docs are still bad even though you sent us a fix. Can you let me know if you still have time to look at this? It's fine if you don't, we can take it over and write the test for it.

Hixie commented:
@goderbauer any chance you can land this for @vpaladino778? i feel really bad about making him jump through all these hoops just for a simple typo fix.

goderbauer committed:
review and add test

vpaladino778 commented:
Hey @Hixie !

Sorry for the delay on this, i tried to write the test but my whole local dev environment was messed up so i wasnt able to get it working. Ya'll are more than welcome to push up the required change to my branch

goderbauer commented:
@vpaladino778 No worries, I just pushed a commit to this branch that resolves the outstanding comments. Once @gspencergoog has reviewed it, this should be ready to land. Thanks for your contribution!

gspencergoog reviewed (approved):
![32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d](https://user-images.githubusercontent.com/8867023/37737237-66e9c9dc-2d10-11e8-8797-13a405eace49.png)

gspencergoog commented:
Thanks for the PR, and the work to get it tested.

goderbauer committed:
Update text_field.0.dart

goderbauer reviewed (approved):
LGTM

goderbauer committed:
Update text_field.0.dart

goderbauer committed:
Update text_field.0.dart

goderbauer commented:
This is now blocked on https://github.com/flutter/flutter/issues/123669.
